libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1447,6 +1465,56 @@ string YulUtilFunctions::readFromStorageDynamic(Type const& _type, bool _splitFu
 		.render();
 	});
 }
+string YulUtilFunctions::readFromStorageReferenceType(Type const& _type, size_t _offset)
```

Comment:
Why didn't you "conversion"?

Reply:
The main difference about value types is that reading clearly always goes to the stack. For reference types, it can either go to memory or storage. It is clear that "from storage" implies "to memory" but this is less clear for "from calldata".

Reply:
Actually it turns out that the conversion function from calldata to memory is already implemented for structs.

Reply:
Ah, just saw that you also modified the conversion function. I will take another look tomorrow!